cmd/list.go

```diff
@@ -82,62 +98,198 @@ func list(cmd *cobra.Command) {
+
+			switch col {
+			case "id":
+				value = workflow.ID
+			case "user":
+				value = workflow.User
+			case "size":
+				if humanReadable {
+					value = workflow.Size.HumanReadable
+				} else {
+					value = workflow.Size.Raw
+				}
+			case "progress":
+				progress := workflow.Progress
+				finishedInfo := displayOptionalFloatField(progress.Finished.Total)
+				totalInfo := displayOptionalFloatField(progress.Total.Total)
+				value = finishedInfo + "/" + totalInfo
+			case "duration":
+				value = displayWorkflowDuration(workflow)
+			case "name":
+				value = name
+			case "run_number":
+				value = runNumber
+			case "created":
+				value = workflow.Created
+			case "started":
+				value = displayOptionalStringField(workflow.Progress.RunStartedAt)
+			case "ended":
+				value = displayOptionalStringField(workflow.Progress.RunFinishedAt)
+			case "status":
+				value = workflow.Status
+			case "session_type":
+				// TODO
+			case "session_uri":
+				// TODO
+			case "session_status":
+				// TODO
+			}
+
+			row = append(row, value)
+		}
+		data = append(data, row)
+	}
+
 	if jsonOutput {
-		utils.DisplayJsonOutput(listResp.Payload)
+		// TODO Fix json output
+		utils.DisplayJsonOutput(data)
 	} else {
-		displayListPayload(listResp.Payload)
+		utils.DisplayTable(header, data)
 	}
 }
 
-func displayListPayload(p *operations.GetWorkflowsOKBody) {
-	header := []interface{}{
-		"NAME",
-		"RUN_NUMBER",
-		"CREATED",
-		"STARTED",
-		"ENDED",
-		"STATUS",
+func displayWorkflowDuration(workflow *operations.GetWorkflowsOKBodyItemsItems0) string {
+	runStartedAt := workflow.Progress.RunStartedAt
+	runFinishedAt := workflow.Progress.RunFinishedAt
+	if runStartedAt == nil {
+		return "-"
+	}
+	startTime := utils.FromIsoToTimestamp(*runStartedAt)
+	var endTime time.Time
+	if runFinishedAt != nil {
+		endTime = utils.FromIsoToTimestamp(*runFinishedAt)
+	} else {
+		endTime = time.Now()
 	}
-	var rows [][]interface{}
+	duration := endTime.Sub(startTime).Round(time.Second).Seconds()
+	return fmt.Sprintf("%g", duration)
+}
 
-	for _, workflow := range p.Items {
-		var row []interface{}
-		workflowNameAndRunNumber := strings.SplitN(workflow.Name, ".", 2)
-		row = append(
-			row,
-			workflowNameAndRunNumber[0],
-			workflowNameAndRunNumber[1],
-			workflow.Created,
-			displayOptionalField(workflow.Progress.RunStartedAt),
-			displayOptionalField(workflow.Progress.RunFinishedAt),
-			workflow.Status,
-		)
-		rows = append(rows, row)
-	}
-
-	utils.DisplayTable(header, rows)
+func getWorkflowNameAndRunNumber(workflowName string) (string, string) {
+	workflowNameAndRunNumber := strings.SplitN(workflowName, ".", 2)
+	if len(workflowNameAndRunNumber) < 2 {
+		return workflowName, ""
+	}
+	return workflowNameAndRunNumber[0], workflowNameAndRunNumber[1]
 }
 
-func displayOptionalField(value *string) string {
+func buildListHeaders(
+	runType string,
+	verbose, includeWorkspaceSize, includeProgress, includeDuration bool,
+) map[string][]any {
```

Comment:
What do you think about returning only the `[]string` headers, instead of the map? The type is already known, so we do not need to return the headers for both `batch` and `interactive`.

PS: I think it is better to use `string` instead of `any` here!

Reply:
Makes sense. As for the `any`, it's the same problem as the comment above